libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinGradleBuildServices.kt

```diff
@@ -72,57 +103,11 @@ internal class KotlinGradleBuildServices private constructor(
     }
 
     override fun buildFinished(result: BuildResult) {
```

Comment:
It looks like configuration cached runs still rely on this method being invoked (please correct me if I'm wrong). If you'd like to be notified when build finished, build services provide you with such lifecycle (see my comment on `KotlinGradleBuildListener`).